staging/src/k8s.io/apimachinery/pkg/util/net/interface.go

```diff
@@ -411,6 +412,17 @@ func chooseHostInterfaceFromRoute(routes []Route, nw networkInterfacer, addressF
 				klog.V(4).Infof("Found active IP %v ", finalIP)
 				return finalIP, nil
 			}
+			// In case of network setups where default routes are present, but network
+			// interfaces use only link-local addresses (e.g. as described in RFC5549).
+			// the global IP is assigned to the loopback interface
+			loopbackIP, err := getIPFromInterface(LoopbackInterfaceName, family, nw)
```

Comment:
Do we need to make a similar check for loopback addresses in `chooseIPFromHostInterfaces`? Currently it also skips the loopback interface:

https://github.com/kubernetes/kubernetes/blob/078d2e65b5b654c9b2c8a6141a853290856a57fa/staging/src/k8s.io/apimachinery/pkg/util/net/interface.go#L296-L299

Reply:
Ah I see now -- this is only relevant when there's a default route and `chooseIPFromHostInterfaces` is not called unless there are no default routes anyways.